sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCountMergePartialFn.java

```diff
@@ -44,109 +37,55 @@ private HllCountMergePartialFn() {}
-   */
-  private static final class HyperLogLogPlusPlusWrapperCoder<T>
-      extends AtomicCoder<HyperLogLogPlusPlusWrapper<T>> {
-
-    private static final HyperLogLogPlusPlusWrapperCoder<?> INSTANCE =
-        new HyperLogLogPlusPlusWrapperCoder<>();
-
-    private final Coder<HyperLogLogPlusPlus<T>> nullableHllCoder =
-        NullableCoder.of(HyperLogLogPlusPlusCoder.of());
-
-    // Generic singleton factory pattern; the coder works for all HyperLogLogPlusPlusWrapper objects
-    // at runtime regardless of type T
-    @SuppressWarnings("unchecked")
-    static <T> HyperLogLogPlusPlusWrapperCoder<T> of() {
-      return (HyperLogLogPlusPlusWrapperCoder<T>) INSTANCE;
-    }
-
-    @Override
-    public void encode(HyperLogLogPlusPlusWrapper<T> value, OutputStream outStream)
-        throws IOException {
-      nullableHllCoder.encode(value.hll, outStream);
-    }
-
-    @Override
-    public HyperLogLogPlusPlusWrapper<T> decode(InputStream inStream) throws IOException {
-      return new HyperLogLogPlusPlusWrapper<T>(nullableHllCoder.decode(inStream));
-    }
-
-    @Override
-    public void verifyDeterministic() throws NonDeterministicException {
-      nullableHllCoder.verifyDeterministic();
-    }
-  }
-
   @Override
-  public Coder<HyperLogLogPlusPlusWrapper<T>> getAccumulatorCoder(
+  public Coder<HyperLogLogPlusPlus<T>> getAccumulatorCoder(
       CoderRegistry registry, Coder<byte[]> inputCoder) {
-    return HyperLogLogPlusPlusWrapperCoder.of();
+    return NullableCoder.of(HyperLogLogPlusPlusCoder.of());
   }
 
   @Override
-  public HyperLogLogPlusPlusWrapper<T> createAccumulator() {
-    return new HyperLogLogPlusPlusWrapper<>(null);
+  public HyperLogLogPlusPlus<T> createAccumulator() {
+    return null;
   }
 
   @Override
-  public HyperLogLogPlusPlusWrapper<T> addInput(
-      HyperLogLogPlusPlusWrapper<T> accumulator, byte[] input) {
-    if (accumulator.hll == null) {
+  public HyperLogLogPlusPlus<T> addInput(HyperLogLogPlusPlus<T> accumulator, byte[] input) {
+    if (accumulator == null) {
       @SuppressWarnings("unchecked")
-      HyperLogLogPlusPlus<T> hll = (HyperLogLogPlusPlus<T>) HyperLogLogPlusPlus.forProto(input);
-      accumulator.hll = hll;
+      HyperLogLogPlusPlus<T> result = (HyperLogLogPlusPlus<T>) HyperLogLogPlusPlus.forProto(input);
+      return result;
     } else {
-      accumulator.hll.merge(input);
+      accumulator.merge(input);
+      return accumulator;
     }
-    return accumulator;
   }
 
   @Override
-  public HyperLogLogPlusPlusWrapper<T> mergeAccumulators(
-      Iterable<HyperLogLogPlusPlusWrapper<T>> accumulators) {
-    HyperLogLogPlusPlusWrapper<T> merged = createAccumulator();
-    for (HyperLogLogPlusPlusWrapper<T> accumulator : accumulators) {
-      if (accumulator.hll == null) {
+  public HyperLogLogPlusPlus<T> mergeAccumulators(Iterable<HyperLogLogPlusPlus<T>> accumulators) {
+    HyperLogLogPlusPlus<T> merged = createAccumulator();
```

Comment:
They are functionally equivalent. Calls `createAccumulator()` because basically every implementation of `mergeAccumulators()` will start with that (so it's like a paradigm?). Would leave it as it is.